tests/devices/qubit_mixed/test_qubit_mixed_apply_operation.py

```diff
@@ -0,0 +1,611 @@
+        init_state = np.reshape(init_state, shape_state)
+        target_state = np.reshape(x[2], shape_state)
+        res = apply_method(op, init_state)
+
+        assert np.allclose(res, target_state, atol=tol, rtol=0)
+
+    x_apply_channel_mixed = [
+        [1, PauliX(wires=0), max_mixed_state(1)],
+        [2, Hadamard(wires=0), max_mixed_state(2)],
+        [2, CNOT(wires=[0, 1]), max_mixed_state(2)],
+        [2, ISWAP(wires=[0, 1]), max_mixed_state(2)],
+        [
+            1,
+            AmplitudeDamping(0.5, wires=0),
+            np.array([[0.75 + 0.0j, 0.0 + 0.0j], [0.0 + 0.0j, 0.25 + 0.0j]]),
+        ],
+        [
+            1,
+            DepolarizingChannel(0.5, wires=0),
+            np.array([[0.5 + 0.0j, 0.0 + 0.0j], [0.0 + 0.0j, 0.5 + 0.0j]]),
+        ],
+        [
+            1,
+            ResetError(0.1, 0.5, wires=0),
+            np.array([[0.3 + 0.0j, 0.0 + 0.0j], [0.0 + 0.0j, 0.7 + 0.0j]]),
+        ],
+        [1, PauliError("Z", 0.3, wires=0), max_mixed_state(1)],
+        [2, PauliError("XY", 0.5, wires=[0, 1]), max_mixed_state(2)],
+    ]
+
+    @pytest.mark.parametrize("x", x_apply_channel_mixed)
+    def test_channel_mixed(self, x, tol, apply_method):
+        """Tests that channels are correctly applied to the maximally mixed state"""
+        nr_wires = x[0]
+        op = x[1]
+        shape_state = [QUDIT_DIM] * 2 * nr_wires
+        init_state = np.reshape(max_mixed_state(nr_wires), shape_state)
+        target_state = np.reshape(x[2], shape_state)
+        res = apply_method(op, init_state)
+
+        assert np.allclose(res, target_state, atol=tol, rtol=0)
+
+
+@pytest.mark.parametrize("ml_framework", ml_frameworks_list)
+class TestBroadcasting:  # pylint: disable=too-few-public-methods
+    """Tests that broadcasted operations are applied correctly."""
+
+    broadcasted_ops = [
+        qml.RX(np.array([np.pi, np.pi / 2, np.pi / 4]), wires=2),
+        qml.PhaseShift(np.array([np.pi, np.pi / 2, np.pi / 4]), wires=2),
+        qml.IsingXX(np.array([np.pi, np.pi / 2, np.pi / 4]), wires=[1, 2]),
+        qml.QubitUnitary(
+            np.array([unitary_group.rvs(8), unitary_group.rvs(8), unitary_group.rvs(8)]),
+            wires=[0, 1, 2],
+        ),
+    ]
+
+    unbroadcasted_ops = [
+        qml.PauliX(2),
+        qml.PauliZ(2),
+        qml.CNOT([1, 2]),
+        qml.RX(np.pi, wires=2),
+        qml.PhaseShift(np.pi / 2, wires=2),
+        qml.IsingXX(np.pi / 2, wires=[1, 2]),
+        qml.QubitUnitary(unitary_group.rvs(8), wires=[0, 1, 2]),
+    ]
+
+    @pytest.mark.parametrize("op", broadcasted_ops)
+    def test_broadcasted_op(self, op, ml_framework):
+        """Tests that batched operations are applied correctly to an unbatched state."""
+        num_q = 3
+        state = math.asarray(get_random_mixed_state(num_q), like=ml_framework)
+
+        res = apply_operation(op, state)
+
+        expanded_mat = TestOperation.expand_matrices(op, 3, batch_size=3)
```

Comment:
I don't think it makes sense to reference class methods of `TestOperation` here. If the functions are needed for tests outside the `TestOperation` class, then they probably shouldn't be inside the `TestOperation` class.